graylog2-web-interface/src/components/configurationforms/ConfigurationForm.jsx

```diff
@@ -153,9 +153,12 @@ const ConfigurationForm = createReactClass({
       shouldAutoFocus = false;
     }
 
-    const configFieldKeys = $.map(this.state.configFields, (v, k) => k).sort(this._sortByOptionality);
+    const configFieldKeys = $.map(this.state.configFields, (v, k) => k)
+      .map((v, i) => ({ key: v, pos: i }))
```

Comment:
I know you didn't wrote that line, but I would vote for replacing the `$.map(this.state.configFields, (v, k) => k)` with the much clearer, in my opinion, `Object.keys(this.state.configFields)`.

Please also use a clearer name for `v`. I suggest `key` instead. In the end we minify the code, so better make it easier to understand.